.github/workflows/function-composition.yml

```diff
@@ -361,3 +391,93 @@ jobs:
         if: ${{ always() }}
         run: |
           kn service delete --all --namespace default --wait
+
+  bench-stacking-training-compose:
+    name: Stacking Training (docker-compose)
+    needs: push-stacking-training
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        transfer-type: [ s3 ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: 'true'
+
+      - name: start multi-container program with docker-compose
+        env:
+          AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+          AWS_SECRET_KEY: ${{ secrets.AWS_SECRET_KEY }}
+        run: |
+          cd ./function-images/tests/stacking-training/
+          docker-compose -f docker-compose-${{ matrix.transfer-type }}.yml up &> log_file &
+          sleep 1m
+      - name: invoke the chain
+        run: |
+          ./bin/grpcurl -plaintext localhost:50051 helloworld.Greeter.SayHello
+      - name: show docker-compose log
+        if: ${{ always() }}
+        run: cat  ./function-images/tests/stacking-training/log_file
+
+  bench-stacking-training-knative:
+    name: Video analytics benchmark (Knative)
+    needs: push-stacking-training
+    env:
+      GOCACHE: /root/tmp/gocache
+      GOPATH: /root/tmp/gopath
+      TMPDIR: /root/tmp
+    runs-on: [ stock-knative ]
+    strategy:
+      fail-fast: false
+      #matrix:
+        #transfer-type: [ s3 ]
+    steps:
+      - name: Setup TMPDIR
+        run: mkdir -p $TMPDIR
+
+      - uses: actions/checkout@v2
+        with:
+          lfs: 'true'
+
+      - name: Deploy functions
+        env:
+          AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+          AWS_SECRET_KEY: ${{ secrets.AWS_SECRET_KEY }}
+        run: |
+          ./function-images/tests/kn_deploy.sh ./function-images/tests/stacking-training/manifests/service*
```

Comment:
these functions, particularly the driver, needs some time to initialize (e.g., generate the dataset). As a result, the invoker's request times out.

Please add a sleep for 1min after the deployment